app/Models/Extensions/AlbumBuilder.php

```diff
@@ -87,10 +93,22 @@ public function getModels($columns = ['*']): array
 
 			$maxTsSelect = $minTsSelect->clone()->select(DB::raw('MAX(taken_at)'));
 
-			$this->addSelect([
-				'min_taken_at' => $minTsSelect,
-				'max_taken_at' => $maxTsSelect,
-			]);
+			if (Configs::getValueAsBool('show_num_albums')) {
```

Comment:
No, this would make the attributes of the object change depending of the config. This significantly complexify the code.

Just have this as a config sent to the front-end and act on it then, not in the back end. It is not hiding something sensitive as the user actually see the number of sub albums and would also be able to open said album and count.

Reply:
This should not have been there and has since been fixed by changing the settings from 'show_num_albums' and 'show_num_photos' to the more general and IMHO preferred 'album_decorations' setting. However, I still process the settings in the back-end as an attempt to simplify the queries as much as possible. The attributes `num_subalbums` and `num_photos` are now always returned but just set to `-1` if no query was issued.